applications/ray/kuberay-tpu-webhook/Troubleshooting.md

```diff
@@ -0,0 +1,44 @@
+# Ray TPU Webhook Troubleshooting / FAQ Guide
+Common issues and their solutions when deploying Ray TPU worker groups with the webhook.
+Solutions will be added as new issues are encountered.
+
+## `TPU_WORKER_HOSTNAMES` aren't injected into the Pod environment
+
+### Solution #1
+`TPU_WORKER_HOSTNAMES` are only injected for multi-host worker groups. If you're expecting `TPU_WORKER_HOSTNAMES` to be injected, check that the `NumOfHosts` field in your Ray worker group spec is set to a value greater than 1.
```

Comment:
I think there is another corner case where the TPU_WORKER_IDs could be incorrectly assigned if a delete/recreate with the same Ray cluster name happens too quickly. Maybe suggest creating clusters with unique names or wait a few seconds before retrying.

Reply:
Done.